src/nvim/eval.c

```diff
@@ -12393,36 +12393,40 @@ static void f_jobstart(typval_T *argvars, typval_T *rettv, FunPtr fptr)
 }
 
 // "jobstop()" function
-static void f_jobstop(typval_T *argvars, typval_T *rettv, FunPtr fptr)
+// Returns 0, or -1 on error
+static int f_jobstop(typval_T *argvars, typval_T *rettv, FunPtr fptr)
 {
   rettv->v_type = VAR_NUMBER;
   rettv->vval.v_number = 0;
 
+  int status = 0;  // zero or negative error code
+
   if (check_restricted() || check_secure()) {
-    return;
+    return status;
   }
 
   if (argvars[0].v_type != VAR_NUMBER) {
     // Only argument is the job id
     EMSG(_(e_invarg));
-    return;
+    return status;
   }
 
   Channel *data = find_job(argvars[0].vval.v_number, true);
   if (!data) {
-    return;
+    return status;
   }
 
   const char *error = NULL;
   if (data->is_rpc) {
     // Ignore return code, but show error later.
-    (void)channel_close(data->id, kChannelPartRpc, &error);
+    if (channel_close(data->id, kChannelPartRpc, NULL) == false)) {
+      status = -1;
+    }
   }
   process_stop((Process *)&data->stream.proc);
   rettv->vval.v_number = 1;
-  if (error) {
-    EMSG(error);
-  }
+
+  return status;
```

Comment:
`f_xx` functions are VimL functions (`:help eval.txt`). They return things in `rettv`. See other functions for examples. Try out your changes by `:echo jobstop()` after building. (Or better, write a test in `job_spec.lua`.

Reply:
Hi Justin ) I appreciate you taking the time to give me detailed feedback. Meanwhile I'm still trying to build it on Windows but I'm encountering a couple of errors.

I referred to `f_jobstart()`. Does this mean that the return value is set via `rettv->vval.vnumber`? 

I've removed the comments I inserted, and modified rettv->vval.v_number to -1 if an error message is assigned during `channel_close` instead. Is this the right approach?

Reply:
is there a reason to use -1 instead of 0?

> Does this mean that the return value is set via `rettv->vval.vnumber`?

yes. Getting it to build will be helpful so you can try the changes (and write a test). Perhaps create a ticket with the build log, maybe someone can figure it out.

Reply:
My initial thought was to differentiate the status from the other errors, but I suppose not. Anyway, I managed to build and run it on an Ubuntu VM, so I'll look into lua testing now ) Thanks again for your help.